src/cpp/src/lm_encoding.cpp

```diff
@@ -83,7 +83,7 @@ ov::genai::utils::GenerationFinishInfo get_lm_encoded_results(
     std::vector<SequenceGroup::Ptr> sequence_groups,
     std::optional<ov::Tensor> position_ids,
     KVCacheState& kv_cache_state,
-    std::optional<EmbeddingsModel> m_embedding,
+    std::optional<EmbeddingsModel::Ptr> m_embedding,
```

Comment:
1. Why is it a pointer now?
2. It doesn't make much sense to wrap a pointer with optional. Setting the pointer to null already conveys the meaning of absence.